### PR TITLE
MM-15575 Rename Constants.h to fix naming conflict

### PR DESCRIPTION
**Pull request by hmhealey**
RNN v2 also has a file and type called Constants which conflicts with the one that we use. If calling everything "Mattermost_____" is too wordy, we could consider switching to "MM_____" like how React has "RC____"

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15575

#### Device Information
This PR was tested on: iOS Simulator

### CODE REVIEW

**Comment by enahum** [comment]
If something is conflicting it should not be cause of the file naming, probably the way it is being imported?

**Comment by hmhealey** [comment]
@enahum I think it might be the type name, not the file name, that's actually the problem. I fixed how I was importing react-native-navigation as we discussed, but I'm still getting this error from the linker
```
duplicate symbol _OBJC_CLASS_$_Constants in:
    /Users/harrison/Library/Developer/Xcode/DerivedData/Mattermost-foeonfnoezzanqeqwbyqmkslfnof/Build/Products/Debug-iphoneos/libUploadAttachments.a(Constants.o)
    /Users/harrison/Library/Developer/Xcode/DerivedData/Mattermost-foeonfnoezzanqeqwbyqmkslfnof/Build/Products/Debug-iphoneos/libReactNativeNavigation.a(Constants.o)
duplicate symbol _OBJC_METACLASS_$_Constants in:
    /Users/harrison/Library/Developer/Xcode/DerivedData/Mattermost-foeonfnoezzanqeqwbyqmkslfnof/Build/Products/Debug-iphoneos/libUploadAttachments.a(Constants.o)
    /Users/harrison/Library/Developer/Xcode/DerivedData/Mattermost-foeonfnoezzanqeqwbyqmkslfnof/Build/Products/Debug-iphoneos/libReactNativeNavigation.a(Constants.o)
ld: 2 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
I think we still need to rename the Constants type or give it some sort of namespace to fix the conflict

**Comment by hmhealey** [comment]
We discussed earlier using `MMConstants` since it's shorter, but Apple has said that they reserve two-letter prefixes like `NS`. They recommend combining company and product name, so we could do `MMMConstants`. Thoughts?

**Comment by enahum** [comment]
What about `RNMM`?

**Comment by hmhealey** [comment]
I was thinking `MMM` as in "MatterMost Mobile". If we went for including `RN` in it, I think company name would go first, so I'd prefer `MMRN` over `RNMM`.

Also, we need more acronyms. We need to choose MMM, MMRN, or RNMM so that I can upgrade the version of RNN used in MM's RN app 😛

**Comment by enahum** [comment]
Hahahaha `MMM` sounds good
